### DISCUSSION TITLE
Does Arc web browser violate license of uBO?

### DISCUSSION THREAD

**Discussion by piquark6046** [uBlockOrigin/uBlock-issues#2581] [General]
Their website: https://arc.net/

References
 - https://github.com/AdguardTeam/AdguardBrowserExtension/issues/2316#issuecomment-1495746981
 - https://listauthorschat.slack.com/archives/C010PK19W78/p1680616800629959

**Comment by gorhill** [comment]
I was willing to check that browser in a VM but after I click _"Try it for yourself"_, it asks for my email. Declined. Anyway, I think the question is simply whether uBO is just pre-installed, or made built-in in the browser.

**Reply by mapx-** [reply]
for macOS
`https://releases.arc.net/release/Arc-latest.dmg`

from
https://www.reddit.com/r/browsers/comments/y4axee/link_to_download_arc_browser/

**Reply by MasterKia** [reply]
Regardless of whether the browser has uBO pre-installed or built-in, it would need to comply with section 6 of the GPLv3. This section states that if a program incorporates code licensed under the GPL, then the entire program must also be licensed under the GPL. Therefore, if the browser includes code from uBO, the browser itself would have to be licensed under the GPL as well.

**Reply by gorhill** [reply]
I wouldn't mind pre-installed, people can always remove it the same way any extension can be removed. I would mind if uBO's code is used as a built-in blocker.

**Reply by MasterKia** [reply]
I doubt they'd go that far. But it would've been a nifty way to avoid the MV3 limitatons.

[How will Arc handle the transition from Manifest v2 to Manifest v3?](https://www.reddit.com/r/arcbrowser/wiki/index/#wiki_how_will_arc_handle_the_transition_from_manifest_v2_to_manifest_v3.3F:)

> The Browser Company has answered many members about this and **_they will likely release a built-in ad-blocker in Arc to work around this shift of manifest version in regard of ad-blocking problematic._**
This is not 100% sure but they're still figuring out how they'll handle the manifest thing.
We don't have much more information about it yet.

**Comment by seia-soto** [comment]
Forwarded some of my messages from Slack to express my opinion:

1) By case, uBO may not be recognized as a part of web browser itself. Because it is ‘extension’ which is independent to the browser. Also, my current primary browser, LibreWolf, also shipped with uBO pre-installed.

2) Unlike browsers allowing control on adblockers via their own interface like Samsung Internet and Microsoft Edge, the uBO pre-installed on Arc or LibreWolf is not controllable through the browser’s own interface. Be attention at the point: user is not restricted to remove or modify the extension as usual they use Chrome or Chromium based browsers and the users can reinstall it (exact same code and exact same metas) from Chrome Web Store.

3) Installing or pre-installing extensions to browser is not difficult task or requires source code changes.

4) I still don’t agree to your opinion. Also, FSF describes that if the softwares are “aggregated” instead of “combined”, it should not be the violence in the FAQ page of GPL. You should take care before going so on. The main reason I don’t think it’s a violence is same as above. Arc didn’t implement any special code to highly integrate with uBlock Origin even they’ll keep the support of MV2 extensions in the browser. It’s not kind of special interaction with uBlock Origin. The patch or any source code modification of browser to support MV2 extensions and Chromium browser extensions is completely available for all type of browser extensions making the fact that Arc and uBlock Origin is “aggregated” software. This fact can be verified in other ways too: The user can re-install the exact same uBlock Origin extension from Chrome Web Store on Arc browser.

**Reply by piquark6046** [reply]
I hope that there are no strong interactions in the preinstalled uBO.

**Reply by seia-soto** [reply]
If you’ve used the Arc browser at least once, you’ll just understand what I am saying. The way that browser extension works is completely same in Chromium, Google Chrome, and any other Chromium based browsers. It’s not a kind of hope or possibility. It’s fact that pre-installing the browser extension itself is not a kind of violent.

**Reply by seia-soto** [reply]
Additional references:
- https://www.reddit.com/r/ArcBrowser/comments/10oydq6/comment/j6jnfyd/?utm_source=share&utm_medium=web2x&context=3
- https://www.reddit.com/r/ArcBrowser/comments/10oydq6/comment/j6hu9np/?utm_source=share&utm_medium=web2x&context=3

I’ll show you a detailed screenshot that controlling uBlock Origin via Arc browser’s own interface is not implemented after getting MBP from Apple for fix.

**Reply by MasterKia** [reply]
At the bare minimum, they _**are**_ distributing uBO. So they must inform the recipients that the software is licensed under the GPL and provide them with a copy of the license.

> You may convey a work based on the Program, or the modifications to produce it from the Program, in the form of source code under the terms of section 4, provided that you also meet all of these conditions:
a) The work must carry prominent notices stating that you modified it, and giving a relevant date.
**b) The work must carry prominent notices stating that it is released under this License** [...]

I see no mention of that on their website.

**Reply by MasterKia** [reply]
> Arc didn’t implement any special code to highly integrate with uBlock Origin

They **can't** "integrate" GPL licensed code in their proprietary software.

**Reply by Yuki2718** [reply]
> 
> They **can't** "integrate" GPL licensed code in their proprietary software.

Unless they made the whole package also GPL. In any case, license violation is subject to prosecution only on complaint by victim. So it's over when gorhill said he doen't care.

**Reply by MasterKia** [reply]
> license violation is subject to prosecution only on complaint by ~~victim~~ copyright holder(s)

Any contributor _can_ also make a complaint since they're the ones holding the copyright of their own contribution (be it trivial or non-trivial, assuming that that their contributed code is still in effect and being used).

uBO's contributors: 
https://github.com/gorhill/uBlock/graphs/contributors

uBO's filterlists' contributors: 
https://github.com/uBlockOrigin/uAssets/graphs/contributors

**Reply by seia-soto** [reply]
No, any people can integrate GPL licensed program when it is subjected to ‘aggregated’ program. This is why I mention the term ‘aggregate’ in the whole messages. The real purpose of GPL is not forcing people to disclose the source by including GPL licensed program.

I would say it’s clear violation if Arc added unique support for uBlock Origin. However, the support for manifest v2 and ad blockers are not specific support to uBlock Origin, and this fact makes Arc to be freed from licensing problem. This is literally not depends on Arc’s distribution of uBlock Origin.

**Reply by Yuki2718** [reply]
I'd say 'integrate' is not allowed, while 'aggregated probram' or 'package with' is okay. The point is that the GPL program must be separated from the proprietary part. It's no problem if uBO is pre-installed just as a Chromium extension which user can disable or uninstall. If there's more interaction, it matters.

**Reply by seia-soto** [reply]
Yes, I have a same point with you. It is not 'integrated' as you say. It's just 'aggregation' as a browser extension. I posted some example of integration to the Slack channel such as Microsoft Edge with ABP.
